### PR TITLE
feat: cortex ps

### LINKED ISSUE #1187: epic: `cortex ps` and Process Status
## Goal 

- `cortex ps` lists running models and free/used memory
- We should ideally be able to show VRAM and RAM use (free vs. total)

## Key Questions
- How do we do this in a stateless manner? (or persist a `*.lock` to Cortex data folder?)

## Linked PRs
- [x] https://github.com/janhq/cortex.cpp/pull/1363
- [x] https://github.com/janhq/cortex.cpp/pull/1345

## DX
`cortex start`
- should stdout a link for api playground
- should correctly handle if port is occupied
```sh
20240919 09:30:08.293674 UTC 3351741 INFO  Host: 127.0.0.1 Port: 3928
 - main.cc:32
Server started
```
Current logs a bit confusing

`cortex stop`

`cortex ps`
- if no active processes, should return a msg to that effect

### PR DESCRIPTION
**Pull request by namchuai**
## Describe Your Changes

- Add `cortex ps` command.

## Screenshots
Screenshot 2024-09-26 at 17 59 42

## Fixes Issues

- fixes #1187 

## Self Checklist

- [ ] Added relevant comments, esp in complex areas
- [ ] Updated docs (for bug fixes / features)
- [ ] Created issues for follow-up changes or refactoring needed

### CODE REVIEW

File: engine/utils/string_utils.h [hunk 1]
```diff
@@ -29,4 +31,48 @@ inline std::vector<std::string> SplitBy(const std::string& str,
...
+  milliseconds %= millisInHour;
+
+  uint64_t minutes = milliseconds / millisInMinute;
+  milliseconds %= millisInMinute;
+
+  uint64_t seconds = milliseconds / millisInSecond;
+
+  std::ostringstream oss;
+
+  if (days > 0) {
```
Comment (vansangpfiev) [inline engine/utils/string_utils.h:64]: Data type is uint64_t, so this condition always satisfied?

**Review by nguyenhoangthuan99 — APPROVED** [review]

File: engine/utils/string_utils.h (hunk 1 above)
Comment (namchuai) [reply engine/utils/string_utils.h:64]: hmm, it can still == 0 right?

Comment (vansangpfiev) [reply engine/utils/string_utils.h:64]: ah, you're right, my bad.

**Review by vansangpfiev — APPROVED** [review]

**Comment by namchuai** [comment]
Thanks guys, force push for rebasing

**Pull request merged**